XMonad/Layout/OnHost.hs

```diff
@@ -154,3 +159,17 @@ eqFQDN a b
   | '.' `elem` a                 = takeWhile (/= '.') a ==                    b
   |                 '.' `elem` b =                    a == takeWhile (/= '.') b
   | otherwise                    =                    a ==                    b
+
+-----------------------------------------------------------------------
+-- cbits
+
+foreign import ccall "gethostname" gethostname :: CString -> CSize -> IO CInt
+
+getHostName :: IO (Maybe String)
+getHostName = allocaArray0 size $ \cstr -> do
+  throwErrnoIfMinus1_ "getHostName" $ gethostname cstr (fromIntegral size)
```

Comment:
I was briefly worried here, but since `runLayout` is in a `catchX` statement in the core this should never blow up XMonad

Reply:
No, but we do occasionally have people who get confused because their layout suddenly became `Full`.

Reply:
I guess this shouldn't worsen the problem, since `$HOST` is still checked first, and only if that fails do we call this function

Reply:
After reading the manpage a couple times, I convinced myself this will never fail on Linux. And seems quite unlikely to fail elsewhere too.